File: scripts/sync-to-openmemory.mjs
#!/usr/bin/env node
/**
 * Node.js script to sync memories to OpenMemory using VS Code environment
 * This script processes the JSON batches created by sync-to-openmemory.py
 */

import fs from 'fs';

async function main() {
    const batchFile = '/home/vivi/pixel/memory_sync_batch.json';
    
    if (!fs.existsSync(batchFile)) {
        console.error(`Batch file not found: ${batchFile}`);
        process.exit(1);
    }
    
    const data = JSON.parse(fs.readFileSync(batchFile, 'utf8'));
    console.log(`Found ${data.total_memories} memories in ${data.total_batches} batches`);
    
    const commands = [];
    
    for (const batch of data.batches) {
        for (const memory of batch) {
            const escapedContent = JSON.stringify(memory.content);
            commands.push(`mcp_openmemory_add-memory ${escapedContent}`);
        }
    }
    
    const commandFile = '/home/vivi/pixel/openmemory_sync_commands.txt';
    fs.writeFileSync(commandFile, commands.join('\n'));
    
    console.log(`Created ${commands.length} commands in ${commandFile}`);
    console.log('Commands ready for VS Code MCP execution');
}

main().catch(console.error);

File: scripts/sync-to-openmemory.py
#!/usr/bin/env python3
import json
import sys
from pathlib import Path


def main():
    backup_dir = Path("/home/vivi/pixel/memory_backups")
    backup_files = list(backup_dir.glob("combined_memories_*.json"))

    if not backup_files:
        return 1

    latest_backup = max(backup_files, key=lambda f: f.stat().st_mtime)

    with open(latest_backup) as f:
        memories = json.load(f)

    if not memories:
        return 1

    batches = [memories[i : i + 10] for i in range(0, len(memories), 10)]

    with open("/home/vivi/pixel/memory_sync_batch.json", "w") as f:
        json.dump(
            {"total_memories": len(memories), "total_batches": len(batches), "batches": batches},
            f,
            indent=2,
        )

    return 0


if __name__ == "__main__":
    sys.exit(main())
